src/java/org/apache/cassandra/streaming/StreamManager.java

```diff
@@ -213,6 +226,98 @@ private static double calculateEffectiveRateInBytes(double throughput)
     private final Map<UUID, StreamResultFuture> initiatorStreams = new NonBlockingHashMap<>();
     private final Map<UUID, StreamResultFuture> followerStreams = new NonBlockingHashMap<>();
 
+    private final Map<UUID, StreamingState> states = new NonBlockingHashMap<>();
+    private volatile ScheduledFuture<?> cleanup = null;
+
+    public StreamManager()
+    {
+        addListener(new StreamListener()
```

Comment:
ill move to start

Reply:
ok - make sure that you remove all listeners in `stop()` otherwise when you call `start()` after you call `stop()` the listeners will start piling up.